common/lib/xmodule/xmodule/mongo_utils.py

```diff
@@ -36,6 +36,9 @@ def connect_to_mongodb(
         # No 'replicaSet' in kwargs - so no secondary reads.
         mongo_client_class = pymongo.MongoClient
 
+    # If the MongoDB server uses a separate authentication database that should be specified here
+    auth_source = kwargs.pop('auth_source', '') or None
```

Comment:
Why are you popping it out of the kwargs dict here?  Why not just do a `get` like you did above?

Reply:
Later in this function the `kwargs` are passed directly to `mongo_client_class` which then throws an error since it doesn't understand the `auth_source` keyword. 

I'm using `get` above since there the `kwargs` are not directly passed, and any extra parameters are passed via `extra`. I generally don't like modifying `kwargs` so avoided it there.

Reply:
cool, thanks for the context.